### PR TITLE
Add File tests

### PR DESCRIPTION
**Pull request by GanShun**
Signed-off-by: Gan Shun Lim <ganshun@gmail.com>

### CODE REVIEW

File: uefi/file_test.go [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+	var tests = []struct {
+		buf  []byte
+		msgs []string
+	}{
+		{emptyPadFile, nil},
+		{badFreeFormFile, []string{"file FFFFFFFF-FFFF-FFFF-FFFF-FFFFFFFFFFFF header checksum failure! sum was 54"}},
+		{goodFreeFormFile, nil},
+	}
+	for _, test := range tests {
+		f, err := NewFile(test.buf)
```
Comment (rjoleary) [inline uefi/file_test.go]: Use subtests: `t.Run(test.name, func (t *testing.T) {...})`

File: uefi/file_test.go [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+		newBuf  []byte
+	}{
+		{emptyPadFile, emptyPadFile},
+		{badFreeFormFile, goodFreeFormFile},
+		{goodFreeFormFile, goodFreeFormFile},
+	}
+	// Set erasepolarity to FF
+	Attributes.ErasePolarity = 0xFF
+	for _, test := range tests {
+		f, err := NewFile(test.origBuf)
```
Comment (rjoleary) [inline uefi/file_test.go]: Ditto

File: uefi/section_test.go [hunk 3]
```diff
@@ -12,7 +12,7 @@ var (
 	emptySec     = make([]byte, 0)                                                     // Empty section
 	tinySec      = []byte{4, 0, 0, byte(SectionTypeRaw)}                               // Section header with no data
 	wrongSizeSec = append([]byte{40, 0, 0, byte(SectionTypeRaw)}, make([]byte, 20)...) // Section with a size mismatch
-	smallSec     = append([]byte{24, 0, 0, byte(SectionTypeRaw)}, make([]byte, 20)...) // 24 byte Section
+	smallSec     = append([]byte{22, 0, 0, byte(SectionTypeRaw)}, make([]byte, 18)...) // 24 byte Section
```
Comment (rjoleary) [inline uefi/section_test.go]: The comment still says "24 byte Section".

File: uefi/section_test.go [hunk 4]
```diff
@@ -12,7 +12,7 @@ var (
 	emptySec     = make([]byte, 0)                                                     // Empty section
 	tinySec      = []byte{4, 0, 0, byte(SectionTypeRaw)}                               // Section header with no data
 	wrongSizeSec = append([]byte{40, 0, 0, byte(SectionTypeRaw)}, make([]byte, 20)...) // Section with a size mismatch
```
Comment (rjoleary) [inline uefi/section_test.go]: Can you make a test like this one, but the actual size is too big?

File: uefi/file_test.go [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+		buf []byte
+		msg string
+	}{
+		{emptyFile, "EOF"},
+		{emptyPadFile, ""},
+		{badFreeFormFile, ""},
+		{goodFreeFormFile, ""},
+	}
+	for _, test := range tests {
+		_, err := NewFile(test.buf)
```
Comment (rjoleary) [inline uefi/file_test.go]: ditto

File: uefi/file_test.go [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+	// I don't know how to do it better without rewriting or calling code under test.
+	emptyPadHeader = append(FFGUID[:],
+		[]byte{8, EmptyBodyChecksum, byte(fvFileTypePad), 0, FileHeaderMinLength, 0x00, 0x00, 0xF8}...) // Empty pad file header with no data
+	goodFreeFormHeader = append(FFGUID[:],
+		[]byte{202, EmptyBodyChecksum, byte(fvFileTypeFreeForm), 0, FileHeaderMinLength, 0x00, 0x00, 0xF8}...) // Empty freeform file header with no data
+)
+
+var (
+	// File examples
+	emptyFile        = make([]byte, 0) // Empty file
```
Comment (rjoleary) [inline uefi/file_test.go]: Why not `[]byte{}`?

File: uefi/section_test.go (hunk 4 above)
Comment (GanShun) [reply uefi/section_test.go]: That's not an error but I can make one and check that an error isn't returned

File: uefi/section_test.go (hunk 3 above)
Comment (GanShun) [reply uefi/section_test.go]: Done

File: uefi/file_test.go (hunk 5 above)
Comment (GanShun) [reply uefi/file_test.go]: Done

File: uefi/file_test.go (hunk 2 above)
Comment (GanShun) [reply uefi/file_test.go]: Done

File: uefi/file_test.go (hunk 1 above)
Comment (GanShun) [reply uefi/file_test.go]: I didn't know about this! Thanks for the suggestion!

File: uefi/file_test.go (hunk 6 above)
Comment (GanShun) [reply uefi/file_test.go]: cause I'm dumb. Done

File: uefi/section_test.go (hunk 4 above)
Comment (GanShun) [reply uefi/section_test.go]: Done

**Pull request merged**